components/planner/history/StudentHistoryView/CourseAudit.tsx

```diff
@@ -0,0 +1,611 @@
+  year: number;
+  semester: string;
+  grade: string;
+}
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      width: '100%',
+    },
+    paper: {
+      width: '100%',
+      marginBottom: theme.spacing(2),
+    },
+    table: {
+      minWidth: 750,
+    },
+    visuallyHidden: {
+      border: 0,
+      clip: 'rect(0 0 0 0)',
+      height: 1,
+      margin: -1,
+      overflow: 'hidden',
+      padding: 0,
+      position: 'absolute',
+      top: 20,
+      width: 1,
+    },
+  }),
+);
+
+function descendingComparator<T>(a: T, b: T, orderBy: string) {
+  const bee = '' + b[orderBy]; //casting as string to pass type validation checks
+  const ay = '' + a[orderBy];
+
+  if (orderBy === 'grade') {
+    if (letters[bee] > letters[ay]) {
+      return -1;
+    }
+    if (letters[bee] < letters[ay]) {
+      return 1;
+    }
+    if (bee === 'A+' && ay !== 'A+') {
+      return -1;
+    }
+    if (bee !== 'A+' && ay === 'A+') {
+      return 1;
+    }
+  }
+  if (orderBy === 'semester') {
+    if (bee.split(' ')[0] < ay.split(' ')[0]) {
+      return -1;
+    }
+    if (bee.split(' ')[0] > ay.split(' ')[0]) {
+      return 1;
+    }
+    if (semesters[bee.split(' ')[1]] < semesters[ay.split(' ')[1]]) {
+      return -1;
+    }
+    if (semesters[bee.split(' ')[1]] > semesters[ay.split(' ')[1]]) {
+      return 1;
+    }
+  }
+
+  if (bee < ay) {
+    return -1;
+  }
+  if (bee > ay) {
+    return 1;
+  }
+  return 0;
+}
+
+type Order = 'asc' | 'desc';
+
+function getComparator<Key extends keyof any>(
+  order: Order,
+  orderBy: string,
+): (a: CourseTaken, b: CourseTaken) => number {
+  return order === 'desc'
+    ? (a, b) => descendingComparator(a, b, orderBy)
+    : (a, b) => -descendingComparator(a, b, orderBy);
+}
+
+function stableSort<T>(array: T[], comparator: (a: T, b: T) => number) {
+  const stabilizedThis = array.map((el, index) => [el, index] as [T, number]);
+  stabilizedThis.sort((a, b) => {
+    const order = comparator(a[0], b[0]);
+    if (order !== 0) return order;
+    return a[1] - b[1];
+  });
+  return stabilizedThis.map((el) => el[0]);
+}
+
+interface HeadCell {
+  disablePadding: boolean;
+  id: string;
+  label: string;
+  numeric: boolean;
+}
+
+const headCells: HeadCell[] = [
+  { id: 'catalogCode', numeric: false, disablePadding: true, label: 'Course Name' },
+  { id: 'semester', numeric: true, disablePadding: false, label: 'Semester' },
+  { id: 'grade', numeric: true, disablePadding: false, label: 'Grade' },
+];
+
+interface EnhancedTableProps {
+  classes: ReturnType<typeof useStyles>;
+  numSelected: number;
+  onRequestSort: (event: React.MouseEvent<unknown>, property: string) => void;
+  onSelectAllClick: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  order: Order;
+  orderBy: string;
+  rowCount: number;
+}
+
+function EnhancedTableHead(props: EnhancedTableProps) {
+  const { classes, onSelectAllClick, order, orderBy, numSelected, rowCount, onRequestSort } = props;
```

Comment:
```suggestion
function EnhancedTableHead({
  classes,
  onSelectAllClick,
  order,
  orderBy,
  numSelected,
  rowCount,
  onRequestSort
}: EnhancedTableProps) {
```
You can combine these two lines by destructuring the function parameters in the function header. See [this article](https://medium.com/@rileyhilliard/es6-destructuring-in-typescript-4c048a8e9e15) for more info.

Reply:
Added to the code